### PR TITLE
Fix: ASP.NET Core, enabled=false in appsetting does not disable public Agent API

### PR DESCRIPTION
**Pull request by gregkalapos**
Solves https://github.com/elastic/apm-agent-dotnet/issues/1077.

### The bug

The problem was that in the `ApmMiddlewareExtension.cs` when the agent was disabled, we returned very early and we did not initialize the agent properly. This was fine as long as the public `Agent.Tracer.X` API wasn't used, once it was used, an implicit initialization happened with default values where `enabled=true`, so the agent still created real `Transaction`s and sent those transactions to the APM server.

### The fix

In the `ApmMiddlewareExtension.cs` we make sure `Agent.Setup(components)` is called with the current config. If that config contains `enabled=false`, the agent will always reuse that, so the public API will also return `NoopTransaction` and `NoopSpan`.

### CODE REVIEW

**Comment by gregkalapos** [comment]
I'll need to do another round on this one tomorrow due to the failing test.

File: src/Elastic.Apm.AspNetCore/ApmMiddlewareExtension.cs [hunk 1]
```diff
@@ -52,14 +52,13 @@ public static class ApmMiddlewareExtension
...
-				return builder;
-
 			var config = new AgentComponents(configurationReader: configReader, logger: logger);
 			HostBuilderExtensions.UpdateServiceInformation(config.Service);
 
+			// Agent.Setup must be called, even if agent is disabled. This way static public API usage won't implicitly initialize an agent with default values, instead, this will be reused.
 			Agent.Setup(config);
-			return UseElasticApm(builder, Agent.Instance, logger, subscribers);
+
+			return !configReader.Enabled ? builder : UseElasticApm(builder, Agent.Instance, logger, subscribers);
```
Comment (russcam) [inline src/Elastic.Apm.AspNetCore/ApmMiddlewareExtension.cs:61]: can this just call `return UseElasticApm(builder, Agent.Instance, logger, subscribers);` and let `UseElasticApm(...)` return without setting up middleware?

**Review by russcam — APPROVED** [review]
LGTM 👍  Left a small comment for consideration

File: src/Elastic.Apm.AspNetCore/ApmMiddlewareExtension.cs (hunk 1 above)
Comment (gregkalapos) [reply src/Elastic.Apm.AspNetCore/ApmMiddlewareExtension.cs]: Yes 👍  Makes sense.

**Comment by nikhil-lad-udio** [comment]
enabled=false is app settings does not disable agent API calling in django as well.

**Pull request merged**